src/java.desktop/macosx/classes/sun/lwawt/LWListPeer.java

```diff
@@ -135,17 +136,19 @@ public void removeAll() {
     @Override
     public void select(final int index) {
         synchronized (getDelegateLock()) {
-            getDelegate().setSkipStateChangedEvent(true);
-            getDelegate().getView().setSelectedIndex(index);
-            getDelegate().setSkipStateChangedEvent(false);
+            ListModel<String> model = getDelegate().getModel();
+            if (index >= 0 && index < model.getSize()) {
+                getDelegate().setSkipStateChangedEvent(true);
+                getDelegate().getView().addSelectionInterval(index, index);
+                getDelegate().setSkipStateChangedEvent(false);
```

Comment:
Does it make sense to call `setSkipStateChangedEvent(false)` in a `finally` block? It will ensure the component is left in a consistent state.